deepchem/feat/tests/test_molgan_featurizer.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from deepchem.feat.molecule_featurizers import MolGanFeaturizer
+from deepchem.feat.molecule_featurizers import GraphMatrix
+
+
+class TestMolganFeaturizer(unittest.TestCase):
+
+  def test_featurizer_smiles(self):
+    try:
+      from rdkit import Chem
+    except ModuleNotFoundError:
+      raise ImportError("This method requires RDKit to be installed.")
+
+    smiles = [
+        'C#C[C@@]1(C)[NH2+][CH+]N[C@@H]1C', '[NH-][CH+]Oc1nnon1',
+        'Cn1ncc2c1C=CC2', 'O=C[C@@H]1[C@H]2[C@H]3[C@@H]1[C@H]1[C@@H]2[N@@H+]31',
+        'C#Cc1[nH]cnc1C#N', 'N#C[C@]1(N)CO[C@H]1CO',
+        'C[C@@H]1C(NO)C[C@@H]2O[C@@H]21', 'OC1CC=CC1', 'Cn1c(O)ccc1CO',
+        '[NH-]C1OC[C@@]2(C=O)N[C@@H]12', 'incorrect smiles'
+    ]
+
+    featurizer = MolGanFeaturizer()
+    data = featurizer.featurize(smiles)
+
+    # test featurization
+    valid_graph = list(filter(lambda x: isinstance(x, GraphMatrix), data))
+    invalid_graph = list(filter(lambda x: not isinstance(x, GraphMatrix), data))
+    assert len(data) == len(smiles)
+    assert len(valid_graph) == len(smiles) - 1
+    assert len(invalid_graph) == 1
+
+    # test defeaturization
+    mols = featurizer.defeaturize(data)
```

Comment:
Can you add one more explicit check that the defeaturized molecules' canonical smiles equal the original smiles (canonicalized as well)? That would be a strong test of inversion

Reply:
This will not work as this is very simple featurization that strips chirality and charges of the molecule. I can prepare different set of smiles that are racemic and have no charge. How does it sound?